### PR TITLE
fix: drop ESAPI and transitive commons-configuration 1.x (#1675)

### PR DESCRIPTION
**Pull request by natechadwick-intsof**
## Summary
Closes Dependabot/security tech-debt for unmaintained **commons-configuration 1.x** (sole path: \sapi:2.7.0.0 → commons-configuration:1.10\).

- **SecureStringUtils**: replaced \ESAPI.encoder()\ / \DefaultEncoder\ (HTML/XML/JS/URL/LDAP, URL decode) with \org.owasp.encoder.Encode\ + local RFC 4515-style LDAP hex encoder (ESAPI-compatible escapes).
- Removed \org.owasp.esapi:esapi\ from \perc-security-utils\ and \sitemanage\ (sitemanage had no Java ESAPI imports).
- Removed parent \sapi.version\ / dependencyManagement for ESAPI; dropped obsolete esapi exclusion on \perc-distribution-tree\ → \perc-security-utils\.
- Extended unit tests for LDAP / HTML / XML / JS / URL encoding edge cases; fixed \	estIsHTML\ to call \isHTML\.

**Residual (not in this PR):** dead ESAPI config packaging under \src/main/resources/esapi\ + installer copy to \
xconfig/esapi\ → follow-up **#1800**.

Fixes #1675

## Test plan
- [x] Pre-change: \cd modules/perc-security-utils && ../../mvnw dependency:tree -Dincludes=commons-configuration:commons-configuration\ showed \sapi → commons-configuration:1.10\
- [x] Post-change: same tree empty (no 1.x); sitemanage tree empty for esapi + commons-configuration 1.x
- [x] \cd modules/perc-security-utils && ../../mvnw clean install\ → **BUILD SUCCESS**, Tests run: **301**, Failures: 0
- [x] \cd projects/sitemanage && ../../mvnw clean install\ → **BUILD SUCCESS**
- [x] \cd modules/perc-distribution-tree && ../../mvnw clean install\ → **BUILD SUCCESS**
- [x] Root Spotless: \mvnw.cmd spotless:apply\ **then** \mvnw.cmd spotless:check\ → SUCCESS; only in-scope files committed (out-of-scope Spotless hit on unrelated code-review doc restored)

## Gates evidence
| Gate | Result |
|------|--------|
| Erlang-style self-review | No path I/O changes; LDAP encoder is pure string; behavioral tests added |
| Spotless apply → check | Pass; no monorepo reformat dump |
| Per-module clean install | security-utils, sitemanage, distribution-tree |
| No new warnings attributable to change | Yes (baseline javadoc/analyze noise only) |

## Residual
- #1800 — remove obsolete ESAPI config resources / installer packaging

### CODE REVIEW

File: modules/perc-security-utils/src/main/java/com/percussion/security/SecureStringUtils.java [hunk 1]
```diff
@@ -148,22 +145,85 @@ public static boolean hostMatchesRequest(
...
+    }
+    StringBuilder sb = new StringBuilder(input.length());
+    for (int i = 0; i < input.length(); i++) {
+      char c = input.charAt(i);
+      switch (c) {
+        case '\\':
+          sb.append("\\5c");
+          break;
+        case '/':
+          sb.append("\\2f");
```
[the remark this answers is not in the document]
Comment (natechadwick-intsof) [reply modules/perc-security-utils/src/main/java/com/percussion/security/SecureStringUtils.java:204]: **Mitigation (commit `137ece0fd8`):** Documented intentional `/` to `\2f` escape in `encodeForLdap` / `sanitizeStringForLDAP` Javadoc. This matches ESAPI `DefaultEncoder.encodeForLDAP` (Microsoft ADSI requires forward-slash escaping; RFC 4515 permits it). Behavior unchanged so directory lookups stay ESAPI-compatible; tests still assert `a/b` encodes to `a\2fb`.

File: modules/perc-security-utils/src/test/java/TestSecureStringUtils.java [hunk 2]
```diff
@@ -137,16 +137,66 @@ public void testPwdDecryption() {
...
+  }
+
+  @Test
+  public void testSanitizeForJson() {
+    assertNull(SecureStringUtils.sanitizeForJson(null));
+    assertEquals("plain", SecureStringUtils.sanitizeForJson("plain"));
+    // OWASP JS encoder escapes quotes, angle brackets, and backslashes
+    String encoded = SecureStringUtils.sanitizeForJson("a'b\"c</script>\\");
+    assertNotNull(encoded);
+    assertFalse(encoded.contains("'") && encoded.contains("\""));
```
[the remark this answers is not in the document]
Comment (natechadwick-intsof) [reply modules/perc-security-utils/src/test/java/TestSecureStringUtils.java]: **Mitigation (commit `137ece0fd8`):** Split the combined quote assertion into two independent `assertFalse` checks so failure to escape either single or double quotes fails the test. See `TestSecureStringUtils.testSanitizeForJson`.

File: modules/perc-security-utils/src/main/java/com/percussion/security/SecureStringUtils.java [hunk 3]
```diff
@@ -1278,21 +1338,21 @@ public static String srp(String rp) {
...
-    String ret = "";
-
+    if (s == null) {
+      return null;
+    }
     try {
-      ret = DefaultEncoder.getInstance().encodeForURL(s);
-    } catch (EncodingException e) {
+      return Encode.forUriComponent(s);
+    } catch (RuntimeException e) {
```
[the remark this answers is not in the document]
Comment (natechadwick-intsof) [reply modules/perc-security-utils/src/main/java/com/percussion/security/SecureStringUtils.java:1364]: **Mitigation (commit `137ece0fd8`):** `urlEncode` no longer returns empty string on encoder failure. It logs input length + message (and debug stack) then rethrows the `RuntimeException` so callers are not handed a silently empty URI component. Javadoc updated accordingly.

**Review by vijaya-boddipudi — APPROVED** [review]
